app/models/my_module.rb

```diff
@@ -139,6 +139,10 @@ def self.filter_by_teams(teams = [])
     joins(experiment: :project).where(experiment: { projects: { team: teams } })
   end
 
+  def self.approaching_due_dates
+    where(notification_sent: false).select { |task| task.is_one_day_prior? }
```

Comment:
Don't use .select, try and implement this using a database query (`due_date_at - NOW() < ?...` something like that)